alchemy/src/cloudflare/openauth.ts

```diff
@@ -0,0 +1,450 @@
+import { type Resource, ResourceKind } from "../resource.ts";
+import type { Secret } from "../secret.ts";
+import type { CloudflareApiOptions } from "./api.ts";
+import type { Bindings } from "./bindings.ts";
+import type { KVNamespaceResource } from "./kv-namespace.ts";
+import { Worker, type FetchWorkerProps } from "./worker.ts";
+
+/**
+ * Configuration for an OAuth provider
+ */
+export interface OAuthProviderConfig {
+  /**
+   * OAuth client ID
+   */
+  clientId: Secret;
+
+  /**
+   * OAuth client secret
+   */
+  clientSecret: Secret;
+
+  /**
+   * OAuth scopes to request
+   */
+  scopes?: string[];
+}
+
+/**
+ * Properties for creating an OpenAuth Worker
+ */
+export interface OpenAuthProps<B extends Bindings = Bindings>
+  extends CloudflareApiOptions {
+  /**
+   * Name for the worker
+   * @default id
+   */
+  name?: string;
+
+  /**
+   * OAuth provider configurations
+   */
+  providers: {
+    /**
+     * GitHub OAuth provider configuration
+     */
+    github?: OAuthProviderConfig;
+
+    /**
+     * Google OAuth provider configuration
+     */
+    google?: OAuthProviderConfig;
+
+    /**
+     * Discord OAuth provider configuration
+     */
+    discord?: OAuthProviderConfig;
+
+    /**
+     * Facebook OAuth provider configuration
+     */
+    facebook?: OAuthProviderConfig;
+  };
+
+  /**
+   * KV Namespace for session storage
+   */
+  storage: string | KVNamespaceResource;
+
+  /**
+   * Additional bindings to attach to the worker
+   */
+  bindings?: B;
+
+  /**
+   * Environment variables to attach to the worker
+   */
+  env?: {
+    [key: string]: string;
+  };
+
+  /**
+   * Whether to enable a workers.dev URL for this worker
+   * @default true
+   */
+  url?: boolean;
+
+  /**
+   * TTL configuration for tokens and sessions
+   */
+  ttl?: {
+    /**
+     * Token reuse time in seconds
+     * @default 60
+     */
+    reuse?: number;
+  };
+
+  /**
+   * Custom success handler function for post-authentication logic
+   * This function will be called after successful OAuth authentication
+   *
+   * @param ctx - Authentication context
+   * @param value - Provider response data
+   * @returns User subject data
+   */
+  onSuccess?: (ctx: any, value: any) => Promise<any> | any;
+
+  /**
+   * User schema definition for type safety
+   * Should be a Valibot schema object
+   */
+  subjects?: any;
```

Comment:
Should we install valibot and then use a type-only import for the schema type instead of any?